components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/internal/PreUpdatePasswordActionServiceComponent.java

```diff
@@ -0,0 +1,98 @@
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.identity.action.management.service.ActionConverter;
+import org.wso2.carbon.identity.action.management.service.ActionDTOModelResolver;
+import org.wso2.carbon.identity.certificate.management.service.CertificateManagementService;
+import org.wso2.carbon.identity.user.pre.update.password.action.management.PreUpdatePasswordActionConverter;
+import org.wso2.carbon.identity.user.pre.update.password.action.management.PreUpdatePasswordActionDTOModelResolver;
+
+/**
+ * Service component for the Pre Update Password Action.
+ */
+@Component(
+        name = "pre.update.password.action.service.component",
+        immediate = true
+)
+public class PreUpdatePasswordActionServiceComponent {
+
+    private static final Log LOG = LogFactory.getLog(PreUpdatePasswordActionServiceComponent.class);
+
+    @Activate
+    protected void activate(ComponentContext context) {
+
+        try {
+            BundleContext bundleCtx = context.getBundleContext();
+            bundleCtx.registerService(ActionConverter.class, new PreUpdatePasswordActionConverter(), null);
+            LOG.debug("Pre Update Password Action Converter is enabled");
+
+            bundleCtx.registerService(ActionDTOModelResolver.class, new PreUpdatePasswordActionDTOModelResolver(),
+                    null);
+            LOG.debug("Pre Update Password Action DTO Model Resolver is enabled");
+
+            LOG.debug("Pre Update Password Action bundle is activated");
+        } catch (Throwable e) {
+            LOG.error("Error while initializing Pre Update Password Action service component.", e);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext context) {
+
+        try {
+            BundleContext bundleCtx = context.getBundleContext();
+            bundleCtx.ungetService(bundleCtx.getServiceReference(PreUpdatePasswordActionConverter.class));
+            bundleCtx.ungetService(bundleCtx.getServiceReference(PreUpdatePasswordActionDTOModelResolver.class));
+            LOG.debug("Action management bundle is deactivated");
+        } catch (Throwable e) {
+            LOG.error("Error while deactivating Pre Update Password Action service component.", e);
+        }
+    }
+
+    @Reference(
+            name = "org.wso2.carbon.identity.certificate.management",
+            service = CertificateManagementService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetCertificateManagementService"
+    )
+    private void setCertificateManagementService(CertificateManagementService certificateManagementService) {
+
+        PreUpdatePasswordActionServiceComponentHolder.getInstance()
+                .setCertificateManagementService(certificateManagementService);
+        LOG.debug("CertificateManagementService set in PreUpdatePasswordActionServiceComponentHolder bundle.");
```

Comment:
Let's just use a consistent term to refer to the component and use a consistent way. Something like 'pre password update service component'

Reply:
Fixed